RecoMET/METAlgorithms/src/HcalHaloAlgo.cc

```diff
@@ -128,24 +128,38 @@ HcalHaloData HcalHaloAlgo::Calculate(const CaloGeometry& TheCaloGeometry, edm::H
 	}
     }
 
+
   // Don't use HF.
   int maxAbsIEta = 29;
+
+
+  std::map<int, float> iPhiHadEtMap;
```

Comment:
this is not a data format anymore, to connect with the previous comment.

I'm not sure if an array is more practical: there will have to be another data structure to keep track of the number of towers in a given iPhi (it's not possible to distinguish a tower with 0 hadEnergy from an array entry with zero value).